example-ideal.txt

```diff
@@ -0,0 +1,11 @@
+go left until blocked on left then
+go up until blocked on up and left then
+(search) go down until blocked on down and left then
+go right then
```

Comment:
I might be misunderstanding your syntax, but wouldn't you not want to check 'blocked on left' in the first line of your 'search' loop?